framework/bundles/org.jboss.tools.rsp.secure/src/org/jboss/tools/rsp/secure/crypto/CryptoUtils.java

```diff
@@ -0,0 +1,113 @@
+package org.jboss.tools.rsp.secure.crypto;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.security.InvalidKeyException;
+import java.security.Key;
+import java.security.NoSuchAlgorithmException;
+import java.security.Security;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * A utility class that encrypts or decrypts a file.
+ * 
+ * @author www.codejava.net
+ *
+ */
+public class CryptoUtils {
+	private static final String ALGORITHM = "AES";
+	private static final String TRANSFORMATION = "AES";
+	static {
+		Security.setProperty("crypto.policy", "unlimited");
+	}
+
+	public byte[] keyTo16(byte[] bytes) {
```

Comment:
AES/CBC/PKCS5PADDING  will only pad the to-be-encrypted text. It will not pad the key. The key is expected to be the proper length always. 

When we intend to change the algortithm, we will need to change this code no matter what. It would not make sense to simply allow someone to simply pass in a new encryption algo flag. 

> seems to me the whole file is encoded whereas Eclipse uses values only encoding

This seems very bad practice to me. Encrypting the string "true" or "false" over and over, or specific integers like "1", "2", or any other 1-character value, could (in my opinion) lead to some attacks. It's very possible that rainbow tables exist for AES, especially if the password has ever been released in a hack before. I do not believe it is smart to encrypt only values. 

> provision for algorithm store in the file

This would be 100% useless. In order to know the value, I'd need to decrypt the file first. If I was able to decrypt the file, I'd already know the algorithm used. And if the file is stored in plaintext keys w only encrypted values, we should NOT be storing the algo in it.... ever.  When a hacker finds an encrypted file, they should have no clue what algo was used to generate it, so as to limit their use of rainbow tables. 

> also I don't see a detection of a bad key given to decrypt the file so I think we should store a fixed value if the file to detect such a case

It is there. 

				if (!startsWith(decrypted, magicBytes)) {
					throw new CryptoException("Invalid key", null);
				}

https://github.com/redhat-developer/rsp-server/pull/45/files#diff-1159c38d60d93f67cbd90b054a1df944R44